### PR TITLE
Intial support to Python bindings for the libafl crate

### PR DESCRIPTION
**Pull request by faroukfaiz10**

### CODE REVIEW

File: libafl/src/executors/inprocess.rs [hunk 1]
```diff
@@ -1127,3 +1339,83 @@ mod tests {
...
+                OnDiskCorpus<BytesInput>
+                >,
+            >
+    }
+
+    #[pymethods]
+    impl PythonOwnedInProcessExecutorI32{
+        #[new]
+        fn new(
+            // harness: PyObject,
```
Comment (andreafioraldi) [inline libafl/src/executors/inprocess.rs:1376]: Here look at the python harness in libafl_sugar

**Comment by faroukfaiz10** [comment]
> It may be good to specify the nightly version that works(?)

Sorry for the late reply @domenukk. Actually, at this time, we're still using the stable version after all.

File: bindings/pylibafl/Cargo.toml [hunk 2]
```diff
@@ -5,12 +5,17 @@ edition = "2021"
 
 [dependencies]
 pyo3 = { version = "0.15", features = ["extension-module"] }
-libafl_qemu = { path = "../../libafl_qemu", version = "0.7", features = ["python"] }
+libafl_qemu = { path = "../../libafl_qemu", version = "0.7", features = ["python", "x86_64"] }
```
Comment (domenukk) [inline bindings/pylibafl/Cargo.toml:8]: Could you do multiple pip packages, one for each arch? (or something)

File: fuzzers/baby_fuzzer/baby_fuzzer.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    if type(corpus).__name__ == "OnDiskCorpus":
+        return libafl.Corpus.new_from_on_disk(corpus)
+
+def rand_wrapper(rand):
+    if type(rand).__name__ == "StdRand":
+        return libafl.Rand.new_from_std(rand)
+
+def stage_wrapper(stage):
+    if type(stage).__name__ == "StdScheduledHavocMutationsStageI32":
+        return libafl.StageI32.new_from_std_scheduled(stage)
```
Comment (domenukk) [inline fuzzers/baby_fuzzer/baby_fuzzer.py:33]: Why `I32`?

Comment (faroukfaiz10) [reply fuzzers/baby_fuzzer/baby_fuzzer.py:33]: These wrappers should be shipped with the module (not written by the user). 
This is just a PoC. Ideally, we would have all the types (I8,I16, ..., U64)

Comment (andreafioraldi) [reply fuzzers/baby_fuzzer/baby_fuzzer.py:33]: > Why `I32`?

No generics support on pyo3

**Pull request merged**